packages/web/pages/player/[username].tsx

```diff
@@ -180,12 +180,14 @@ export const getStaticPaths: GetStaticPaths<QueryParams> = async () => {
 export const getStaticProps = async (
   context: GetStaticPropsContext<QueryParams>,
 ) => {
-  const username = context.params?.username;
+  // usernames in the DB are lowercase. This essentially 
+  // makes this page's router variable case-insensitive.
+  const username = context.params?.username?.toLowerCase();
   const player = await getPlayer(username);
```

Comment:
When does this generation happens ? Curious to know where nextjs is aware of the list of players.

Maybe change to `const player = await getPlayer(username) || null;`

And also,  if `username`  is nullish, it is not necessary to call `getPlayer`

Reply:
> When does this generation happens ? Curious to know where nextjs is aware of the list of players.

This is part of next.js's [static rendering](https://nextjs.org/docs/basic-features/pages#static-generation-with-data). At build time, the list of players is fetched. 

> 
> I don't really understand the error, as by reading the code, the player props could not be `undefined` but `null`.

Yeah, I force-pushed a fix for that particular problem.

I changed the player username lookup to be case-insensitive. I believe this will fix the problem.  

Confusingly, in my local DB all the usernames are lowercase as they were all imported via `migrateSourceCredAccounts`, but on prod they are mixed-case.